wgs_somatic-run-wrapper.py

```diff
@@ -0,0 +1,227 @@
+        samples_ready.append(s)
+    return samples_ready
+
+
+def wrapper():
+
+    # using lists to keep track of stuff... could maybe be done in a better way...
+    additional_run_paths = []
+    tumor_samples = []
+    normal_samples = []
+    pair_ids_in_run = []
+    started_samples = []
+
+
+    with open(CONFIG_PATH, 'r') as conf:
+        config = yaml.safe_load(conf)
+
+    # Grab all available local run paths
+    instrument_root_path = config['novaseq']['demultiplex_path']
+    local_run_paths = look_for_runs(instrument_root_path)
+
+    # Read all previously analysed runs
+    previous_runs_file = config['previous_runs_file_path']
+    previous_runs_file_path = os.path.join(ROOT_DIR, previous_runs_file)
+    with open(previous_runs_file_path, 'r') as prev:
+        previous_runs = [line.rstrip() for line in prev]
+
+
+
+    # Loop through each run path and setup Run context class
+    for run_path in local_run_paths:
+        Rctx = RunContext(run_path)
+        # Check if demultiplexing is completed
+        if not Rctx.demultiplex_complete:
+            continue
+
+        # Check if run has been previously analysed
+        if Rctx.run_name in previous_runs:
+            continue
+
+        # Register start time
+        start_time = datetime.now()
+        logger.info(f'Started {Rctx.run_name} at {start_time}')
+
+        # Write run name to previously analysed list to ensure no double-running
+        with open(previous_runs_file_path, 'a') as prev:
+            logger.info(f'Writing {Rctx.run_name} to previous runs list.')
+            print(Rctx.run_name, file=prev)
+
+        # get Rctx and Sctx for current run
+        Rctx_run = Rctx_Sctx_for_run(Rctx)
+
+
+        # Get run paths for samples (or other part of t/n pair) with additional fastqs in other runs
+        for sctx in Rctx_run.sample_contexts:
+            run_paths = get_pair_and_more_fqs(sctx, Rctx.run_tag)
+            if not run_paths == None:
+                run_paths = list(chain.from_iterable(run_paths))
+                for r in run_paths:
+                    additional_run_paths.append(r) if r not in additional_run_paths else additional_run_paths
+
+    # get Rctx and Sctx for additional runs that have samples related to current run
+    # i realized that i don't actually use "additionalRctx" for anything now. but using the function Rctx_Sctx_for_run does append to sample_status which is used below... could be done in a better way if i don't have to use additionalRctx anyway. just appending to sample_status is neccessary.  
+    for run_path in additional_run_paths:
+        additionalRctx = RunContext(run_path)
+        additionalRctx = Rctx_Sctx_for_run(additionalRctx)
+
+
+    # get tumor and normal samples related to current run from approved samples
+    for category, contexts in sample_status.items():
+        if category == 'approved':
+            for Sctx in contexts:
+                if Rctx_run.run_tag == Sctx.sample_id.split("_",1)[1]:
```

Comment:
the run tag check is done since approved samples also contains samples from additional runs. maybe this can be solved in a better way if I don't append samples from additional runs to sample_status. will look at this